src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -730,6 +732,164 @@ public static nuint RotateRight(nuint value, int offset)
 #endif
         }
 
+        /// <summary>
+        /// Accumulates the CRC (Cyclic redundancy check) checksum.
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The data for which to compute the checksum</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static uint Crc32C(uint crc, byte data)
+        {
+            if (Sse42.IsSupported)
+            {
+                return Sse42.Crc32(crc, data);
```

Comment:
Is this call using the correct "Castagnoli algorithm" as you referred to earlier?

Reply:
Yes. See https://en.wikipedia.org/wiki/SSE4 -> SSE4.2